### ISSUE TITLE
PIN keyboard numeric after restore despite alpha PIN

### ISSUE THREAD

**Issue by arikb** [signalapp/Signal-Android#13338]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I have an alphanumeric PIN. I deleted Signal from my phone and re-installed it for reasons. I was pleasantly surprised that I was able to restore my registration with my PIN only, without receiving SMS. It was easy to switch to an alphanumeric keyboard during the entry of my PIN.

However, later on, I got the PIN reminder screen. When I attempted to confirm my PIN, the keyboard was numeric and I was not able to change it to an alphanumeric keyboard. Eventually I worked around it by resetting my PIN (to the same alphanumeric PIN). Since then, when I confirm my PIN the keyboard opens up as alphanumeric.

### Steps to reproduce
- define an alphanumeric PIN
- remove Signal from the device
- install Signal from the device
- register Signal on the device (using the alphanumeric PIN)
- Wait until Signal asks to (periodically) confirm the PIN 
- Tap the prompt to confirm the PIN

**Actual result:** PIN confirmation screen appears with a numeric keyboard, without an option to switch to an alphanumeric keyboard
**Expected result:** PIN confirmation screen appears with an alphanumeric keyboard; OR an option to switch to an alphanumeric keyboard

### Screenshots
I completely forgot to screenshot it when it happened to me. I reinstalled Signal again, I will amend it when I get the reminder message again.

### Device info
**Device:** Oneplus 5T (ONEPLUS A5010)
**Android version:** 9
**Signal version:** 6.42.3

### Link to debug log
I don't think a UI bug will generate a debug log, but I'll grab one as soon as I reproduce.

**Comment by arikb** [comment]
Okay I think I found someone commenting on it on a separate issue - https://github.com/signalapp/Signal-Android/issues/12866#issuecomment-1552099816

The issue doesn't describe my experience but the comment does.

**Comment by arikb** [comment]
Reproduced.

Debug log: https://debuglogs.org/android/6.42.3/37372fda9279ab45e147ee32320c62196af7d5e40e074a50f6af85402f9d1a23

Screenshot from my device (doesn't show the keyboard due to a limitation of the screenshot function):
![signal-2023-12-30-201359](https://github.com/signalapp/Signal-Android/assets/1229613/17520a51-afbe-4b4f-89e7-1fb639cd03dc)


Photo from another device showing the keyboard:
![signal-2023-12-30-201138](https://github.com/signalapp/Signal-Android/assets/1229613/953c1e48-795f-478f-b8f6-9bb17d3dae0e)

**Comment by arikb** [comment]
Hmm... I'm not that familiar with the stale status. Looks like the issue is verified and resulted in a commit - shouldn't it be closed as resolved instead? Or only after QA / testing?

**Comment by nicholas-signal** [comment]
> Hmm... I'm not that familiar with the stale status. Looks like the issue is verified and resulted in a commit - shouldn't it be closed as resolved instead? Or only after QA / testing?

Please close this issue once you confirm that the issue is fixed.

**Comment by arikb** [comment]
Oh, so I'm QA... Now I get it.

I reinstalled 7.0.2, I'm at step 5 in reproducing the bug. It will take 3 days to reproduce.

**Comment by arikb** [comment]
At least in my case, I confirm the issue is fixed.

**Comment by nicholas-signal** [comment]
> Oh, so I'm QA... Now I get it.

We were not able to recreate the conditions exactly, and we didn't want to close the issue without confirming the issue is fixed.

**Comment by arikb** [comment]
No worries, I just didn't understand my role in the process... I fully accept that if I open a ticket for a bug I should be the one to confirm it's resolved.

I received a PIN confirmation request a couple of hours ago and was able to confirm the bug as resolved. I also went over the commit, it looks like it addresses what I experienced.
